manila/share/drivers/netapp/dataontap/client/client_cmode.py

```diff
@@ -2618,21 +2618,60 @@ def set_sis_config(self, volume_name, api_args):
     @na_utils.trace
     def enable_dedup(self, volume_name):
         """Enable deduplication on volume."""
+        interval = 3
+        retries = 5
         api_args = {'path': '/vol/%s' % volume_name}
-        try:
-            self.send_request('sis-enable', api_args)
-        except netapp_api.NaApiError as e:
-            enabled_msg = "has already been enabled"
-            if (e.code == netapp_api.OPERATION_ALREADY_ENABLED and
-                    enabled_msg in e.message):
+
+        @manila_utils.retry(retry_param=exception.ShareBackendException,
+                            interval=interval,
+                            retries=retries,
+                            backoff_rate=1)
+        def try_enable_dedup():
+            try:
+                self.send_request('sis-enable', api_args)
                 return
-            raise e
+            except netapp_api.NaApiError as e:
+                enabled_msg = "has already been enabled"
+                if (e.code == netapp_api.OPERATION_ALREADY_ENABLED and
+                        enabled_msg in e.message):
+                    return
+                active_msg = "sis operation is currently active"
+                if (e.code == netapp_api.OPERATION_ALREADY_ENABLED and
+                        active_msg in e.message):
+                    msg = _('Unable to enable dedup. Will retry the '
+                            'operation. Error details: %s') % e.message
+                    LOG.warning(msg)
+                    raise exception.ShareBackendException(msg=msg)
+                raise e
+
+        try_enable_dedup()
 
     @na_utils.trace
     def disable_dedup(self, volume_name):
         """Disable deduplication on volume."""
+        interval = 3
+        retries = 5
         api_args = {'path': '/vol/%s' % volume_name}
-        self.send_request('sis-disable', api_args)
+
+        @manila_utils.retry(retry_param=exception.ShareBackendException,
+                            interval=interval,
+                            retries=retries,
+                            backoff_rate=1)
+        def try_disable_dedup():
```

Comment:
you could put the retry decorator to the class method, it would be effectively the same

Reply:
I followed _enable_ldap_client() implementation within file.

Reply:
there are extra error handling there, it would be nice that you can fix it. nested functions is unusual pattern

Reply:
done